packages/runtime/src/runtimes/react/controls.tsx

```diff
@@ -153,35 +155,59 @@ type PropsValueProps = {
   children(props: Record<string, unknown>): JSX.Element
 }
 
-// function useResolvedControlValue(
-//   elementData: ReactPage.ElementData,
-//   definition: { type: string },
-//   resourceResolver: ResourceResolver,
-// ): any {
-//   const resolvedValue = resolveControlValue(elementData, definition, resourceResolver)
-//   return useSyncExternalStore(
-//     fn => resolvedValue.subscribe(fn),
-//     () => resolvedValue.readValue(),
-//     () => resolvedValue.readValue(),
-//   )
-// }
-
-// const useResolvedProps = (
-//   propDefs: Record<string, { type: string }>,
-//   elementData: Record<string, ReactPage.ElementData>,
-//   resourceResolver: ResourceResolver,
-// ) => {
-//   const result: Record<string, any> = {}
-
-//   for (const [key, def] of Object.entries(propDefs)) {
-//     const value = useResolvedControlValue(elementData[key], def, resourceResolver)
-//     if (value !== undefined) {
-//       result[key] = value
-//     }
-//   }
-
-//   return result
-// }
+const useResolvedProps = (
```

Comment:
This is the refactored hook that resolves prop values and establishes subscriptions to the underlying resources (via the unified interface). It no longer makes a looped hook call, appropriately responds to prop changes, and correctly unsubscribes when the element changes.